Add pricing plans to vehicle types

mplsmitch opened the pull request:
#### **What problem does your proposal solve? Please begin with the relevant issue number. If there is no existing issue, please also describe alternative solutions you have considered.**
When `system_pricing_plans.json` was extended in v2.2 the `pricing_plan_id` field was added to `free_bike_status.json` to associate pricing plans with vehicles. The thinking was that doing this at the individual vehicle level would allow dynamic pricing based on the vehicle's location.
This poses a challenge for station based systems in that they would be forced to publish the optional `free_bike_status` end point in order to surface pricing information. It also limits vehicles to a single pricing plan.

#### **What is the proposal?**
This proposal adds a `pricing_plans` array to `vehicle_types.json` that contains the plan IDs  of all applicable pricing plans for that vehicle type. It also adds a field that defines a default pricing plan to be used by trip planners in calculating the cost of the a single trip.

The ability to define pricing plans at the individual vehicle level is preserved, which would allow pricing to be set based on vehicle location or other factors. Any pricing plan assigned to individual vehicles in `free_bike_status.json` supersedes the default plan defined in `vehicle_types.json`.

Additional discussion of this can be found in issue #318 

#### **Is this a breaking change?**
- [ ] Yes 
- [x] No
- [ ] Unsure

#### **Which files are affected by this change?**

gbfs.md:

- vehicle_types.json

mplsmitch committed:
Add pricing plans to vehicle types

Adds pricing_plans and default_pricng_plan to vehicle types

richtaylor-ito commented on gbfs.md:
```diff
@@ -395,7 +395,8 @@ Field Name | REQUIRED | Type | Defines
 \- `name` | OPTIONAL | String | The public name of this vehicle type.
-
+\- `default_pricing_plan`| Conditionally REQUIRED | ID | REQUIRED if `system_pricing_plans` and `vehicle_types.json` are defined. A `plan_id` as defined in `system_pricing_plans.json` that identifies a default pricing plan for this vehicle to be used by trip planning applications for purposes of calculating the cost of a single trip using this vehicle type. This default pricing plan is superseded ...
+\- `pricing_plans` | OPTIONAL | Array | Array of all pricing plan IDs as defined in `system_pricing_plans.json` that are applied to this vehicle type. <br /><br />This array SHOULD be published when there are multiple pricing plans defined in `system_prcing_plans.json` that apply to a single vehicle type.
```
Typo, should by `system_pricing_plans.json`

mplsmitch committed:
Fix typo

mplsmitch commented:
I hereby call a vote on this proposal. 

Voting will be open for 10 full calendar days until 11:59PM UTC on September 1st, 2021.

Please vote for or against the proposal, and include the organization for which you are voting in your comment.

Please note if you can commit to implementing the proposal.

kanagy commented:
Google Maps supports this and would like to implement it for docked systems where currently using free_bike_status is unnecessary. It would be nice to have some guidance in the spec to prefer setting default_pricing_plan first for all producers and then override it in free_bike_status if needed.

Nit: For consistency with other fields, would it make sense to rename to `default_pricing_plan_id` and `pricing_plan_ids`?

testower commented:
Entur supports this proposal and will implement it in our aggregation service

richtaylor-ito commented:
Ito World supports this proposal.

nbdh commented:
Nextbike supports this proposal, though we will probably keep publishing `free_bikes_status` and define the `pricing_plan_id` there.

viestat commented:
Dott approves this proposal, however I would argue that `default_pricing_plan` should be optional instead of conditionally required. We will keep on defining on a vehicle level using `free_bikes_status`.

evansiroky commented:
IBI Group supports this proposal, although we agree with @kanagy's suggestion to rename `vehicle_types.json#default_pricing_plan ` to `vehicle_types.json#default_pricing_plan_id` and `vehicle_types.json#pricing_plans` to `vehicle_types.json#pricing_plan_ids` to try to follow a naming convention that shows which fields have IDs as values.

ncancelliere commented:
Spin supports this proposal. We likely will not immediately implement it.

ezmckinn commented:
Superpedestrian supports this proposal. We don't currently have plans to add this to `vehicle_types.json`, and we will continue publishing `pricing_plan_id` within `free_bike_status.json`.

josee-sabourin commented:
Voting is now closed, and it passes! 

Votes in favour:
Google Maps (consumer)
Entur (consumer)
Ito World (producer)
Nextbike (producer)
Dott (producer)
IBI Group (consumer)
Spin (producer)
Superpedestrian (producer)

There we no votes against. 

Regarding @kanagy's and @evansiroky's comments: we will make those semantic changes when we merge into a v2.3-RC.

mplsmitch committed:
update field name and definition

Updates field name default_pricing_plan to default_pricing_plan_id, clarifies definition giving preference to defining default_pricing_plan_id

mplsmitch committed:
Merge branch 'master' into vehicle-pricing

mplsmitch committed:
Add v2.3-RC labels

Add V2.3-RC  label and clarify use of pricing_plan_id

mplsmitch committed:
Merge branch 'vehicle-pricing' of https://github.com/NABSA/gbfs into vehicle-pricing

mplsmitch committed:
Label pricing_plans as v2.3-RC
